src/Symfony/Component/Serializer/Normalizer/NumberNormalizer.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use BcMath\Number;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\NotNormalizableValueException;
+
+/**
+ * Normalizes {@see Number} and {@see \GMP} to a string.
+ */
+final class NumberNormalizer implements NormalizerInterface, DenormalizerInterface
+{
+    public function getSupportedTypes(?string $format): array
+    {
+        return [
+            Number::class => true,
+            \GMP::class => true,
+        ];
+    }
+
+    public function normalize(mixed $data, ?string $format = null, array $context = []): string
+    {
+        if (!$data instanceof Number && !$data instanceof \GMP) {
+            throw new InvalidArgumentException(\sprintf('The data must be an instance of "%s" or "%s".', Number::class, \GMP::class));
+        }
+
+        return (string) $data;
```

Comment:
>InvalidCast: GMP cannot be cast to string

what about psalm's report?
this should use gmp_strval I guess instead?

Reply:
The error is false-positive, a `GMP` object can be casted to a `string` without  issues, even though it's not a documented feature.

But you're right, probably safer to follow the docs

Reply:
I'm fine as is, the stringable behavior is no mistake.

Reply:
I suggest opening an issue on PHP regarding the fact that the `GMP` object implements the casting to string through the internal API overloading cast behavior without implementing `Stringable`. That's the kind of things that PHP has cleaned in recent years for some other classes, so it might be accepted to fix this in the next PHP version (which will then make static analysis tool aware that this class is castable to string without having to maintain a list of special internal classes that are castable to string but not Stringable)